### ISSUE TITLE
Support Ubuntu 16.04 LTS

### ISSUE THREAD

**Issue by gevans** [atomic-penguin/cookbook-nfs#79]
First off, great cookbook! While upgrading to 16.04 I ran into a couple issues though:

* Systemd needs to be selected as the provider for Ubuntu 16.04:

  ```
  ================================================================================
  Error executing action `start` on resource 'service[portmap]'
  ================================================================================
    
  Errno::ENOENT
  -------------
  No such file or directory - /sbin/status
  ```

  I was able to hack around this with:
  ```ruby
  node.default['nfs']['service_provider'] = Hash.new { |h, k| h[k] = Chef::Provider::Service::Systemd }
  ```

* The `statd` service was renamed to `rpc-statd`:

  ```
  ================================================================================
  Error executing action `start` on resource 'service[lock]'
  ================================================================================
    
  Mixlib::ShellOut::ShellCommandFailed
  ------------------------------------
  Expected process to exit with [0], but received '5'
  ---- Begin output of /bin/systemctl --system start statd ----
  STDOUT: 
  STDERR: Failed to start statd.service: Unit statd.service not found.
  ---- End output of /bin/systemctl --system start statd ----
  Ran /bin/systemctl --system start statd returned 5
  ```

  I was able to hack around this with:
  ```ruby
  node.default['nfs']['service']['lock'] = 'rpc-statd'
  ```

Other than that, everything else works fine. I hope we can get a real fix for this soon. :smile:

**Comment by eric-tucker** [comment]
Can't believe this still hasn't been resolved.  It's been nearly six months.  :(

**Comment by atomic-penguin** [comment]
@eric-tucker this was added in 2.2.9 on August 11.  I haven't heard any follow-up issues pertaining to Ubuntu >= 15.04 since then.

**Comment by eric-tucker** [comment]
I added 2.2.11 to my Berksfile, ran with "nfs" in my run list and got issues.  I applied the above fixed provided by @gevans and things worked again.  I assumed that this support was still missing.

**Comment by atomic-penguin** [comment]
@eric-tucker which part the service provider mapping, lock service mapping, or both, had to be corrected on 2.2.11?

I would think this block would select both of those items correctly for 16.04, but perhaps not?

```
if Chef::VersionConstraint.new('>= 15.04').include?(node['platform_version'])
      default['nfs']['service_provider']['idmap'] = Chef::Provider::Service::Systemd
      default['nfs']['service_provider']['portmap'] = Chef::Provider::Service::Systemd
      default['nfs']['service_provider']['lock'] = Chef::Provider::Service::Systemd
      default['nfs']['service']['lock'] = 'rpc-statd'
      default['nfs']['service']['idmap'] = 'nfs-idmapd'
    end
```

**Comment by eric-tucker** [comment]
I'm not sure why it wasn't working but now it seems to be working without setting the additional attributes.  All good now it seems.

Thanks and sorry for the confusion!
